WorldModel/WorldModel/Core/CoreOutput.aslx

```diff
@@ -874,14 +874,16 @@
 
   <function name="Log" parameters="text">
     <![CDATA[
-    //request (Log, text)
+    request (Log, text)
     // Replacing double quotes with 2 single quotes
     text = Replace(text, "\"", "''")
     // Changing syntax from single quotes to escaped double quotes to allow single quotes in log entries
     if (not GetBoolean(game, "nohtmllog")){
       JS.eval("if(typeof(addLogEntry)===\"function\"){ addLogEntry(\""+text+"\"); };")
     }
-    JS.eval("if(!webPlayer && typeof(WriteToLog)===\"function\"){var s = \""+text+"\";WriteToLog(s);}")
+    if (GetBoolean(game, "writelogtofile") and StartsWith(GetFileURL("foo"),"quest")){

```

Comment:
What's with `GetFileURL("foo")` here?

Reply:
Checking for the desktop player without using `ASLEvent`.

Reply:
I feel there must be an easier way... I also feel I saw something had been added that already did something like that?

Reply:
In any case I wonder if there's a way we can make this agnostic of where it's running.

Reply:
> I also feel I saw something had been added that already did something like that?

You're probably thinking of `JS.whereAmI()`, and it uses `ASLEvent`.

> In any case I wonder if there's a way we can make this agnostic of where it's running.

The transcript is missing a JS function online in 5.8, and I'm beginning to notice I doubled up on trying to be sure that didn't happen again.

I could just put a "dummy" `WriteToLog` in **player.js** (which does nothing but exist), then remove all this online checking (and the `eval`, which was just a second backup for that) like this:
```js
if (GetBoolean (game, "writelogtofile")) {
      JS.writeToLog(s)
```